python-backend/app/api/variances/resources/variance_resource.py

```diff
@@ -43,24 +60,107 @@ def get(self):
             'total': pagination_details.total_results,
         }
 
+    def _apply_filters_and_pagination(self, args):
 
-    def _apply_filters_and_pagination(self,
-                                      page_number=PAGE_DEFAULT,
-                                      page_size=PER_PAGE_DEFAULT,
-                                      application_status=None):
         status_filter_values = list(map(
             lambda x: x.variance_application_status_code,
             VarianceApplicationStatusCode.active()))
 
-        if application_status is not None:
-            status_filter_values = application_status.split(',')
+        if args["page_number"] is None:
+            args["page_number"]=PAGE_DEFAULT
+        if args["page_size"] is None:
+            args["page_size"] = PER_PAGE_DEFAULT
 
-        filtered_query = apply_filters(
-            Variance.query.order_by(desc(Variance.received_date)),
-            [{
+        conditions = []
+        if args["application_status"] is not None:
+            status_filter_values = args["application_status"].split(',')
+            conditions.append({
+                'model': 'Variance',
                 'field': 'variance_application_status_code',
                 'op': 'in',
                 'value': status_filter_values
-            }])
+            })
+        if args["compliance_codes"] is not None:
+            compliance_codes_values = args["compliance_codes"].split(',')
+            conditions.append({
+                'model': 'Variance',
+                'field': 'compliance_article_id',
+                'op': 'in',
+                'value': compliance_codes_values
+             })
+
+        if args["expiry_date_before"] is not None:
+            conditions.append({
+              'model': 'Variance',
+              'field':  'expiry_date',
+              'op': '<=',
+              'value': args["expiry_date_before"]
+            })
+        if args["expiry_date_after"] is not None:
+            conditions.append({
+              'model': 'Variance',
+              'field': 'expiry_date',
+              'op': '>=',
+              'value': args["expiry_date_after"]
+            })
+        if args["issue_date_before"] is not None:
+            conditions.append({
+              'model': 'Variance',
+              'field': 'issue_date',
+              'op': '<=',
+              'value': args["issue_date_before"]
+            })
+        if args["issue_date_after"] is not None:
+            conditions.append({
+              'model': 'Variance',
+              'field': 'issue_date',
+              'op': '>=',
+              'value': args["issue_date_after"]
+            })
+
+        if args["major"] is not None:
+            conditions.append({
+                'model': 'Mine',
+                'field': 'major_mine_ind',
+                'op': '==',
+                'value': args["major"]
+            })
```

Comment:
I think you can reduce the bloat on these in a couple different ways. You could either:
1. Refactor this to use the query object directly. Something like:
```python
if args["major"] is not None:
    conditions.append(Mine.major_mine_ind=args["major"])
```
or
2. Abstract this into a separate method (I'm calling it `_build_filter` here) that's invoked something like:
```python
if args["major"] is not None:
    conditions.append(cls._build_filter('Mine', 'major_mine_ind', '==', args))
```

I prefer option 1, personally. If you go with 2, you'll want to name the args the same as the columns, which is probably better for readability anyway. It would look something like:
```python
@classmethod 
def _build_filter(cls, model, field, op, args):
    return {
                'model': model,
                'field': field,
                'op': op,
                'value': args[field]
    }
```

Reply:
I've implemented and tested this idea